### PR TITLE
Remove extra naive pipelines

### LINKED ISSUE #4152: Remove Extra Naive Pipelines from AutoML Search Process
Currently, we are running extra feature selection and naive pipelines during our AutoML search process. This issue tracks removing these extra pipelines while maintaining the same level of performance in AutoML search.

### PR DESCRIPTION
**Pull request by christopherbunn**
Resolves #4152

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: docs/source/user_guide/timeseries.ipynb [hunk 1]
```diff
@@ -540,6 +540,7 @@
     "    y_train,\n",
     "    problem_type=\"time series regression\",\n",
     "    max_batches=1,\n",
+    "    automl_algorithm=\"iterative\",\n",
```
Comment (jeremyliweishih) [inline docs/source/user_guide/timeseries.ipynb:543]: did you add this for `allowed_model_families` to work?

**Review by jeremyliweishih — APPROVED** [review]
LGTM

File: docs/source/user_guide/timeseries.ipynb (hunk 1 above)
Comment (christopherbunn) [reply docs/source/user_guide/timeseries.ipynb:543]: Yep, previously it only just ran the first batch which didn't get great results

**Review by eccabay — APPROVED** [review]

**Pull request merged**